pkg/controller/submariner/submariner_controller.go

```diff
@@ -193,39 +170,32 @@ func (r *ReconcileSubmariner) reconcileRouteagentDaemonSet(instance *submarinerv
 	return nil
 }
 
-func newDeploymentForCR(cr *submarinerv1alpha1.Submariner) *appsv1.Deployment {
-
+func newEngineDaemonSet(cr *submarinerv1alpha1.Submariner) *appsv1.DaemonSet {
 	labels := map[string]string{
 		"app":       "submariner-engine",
 		"component": "engine",
 	}
 
-	replicas := int32(1)
 	revisionHistoryLimit := int32(5)
-	progressDeadlineSeconds := int32(600)
 
-	maxSurge := intstr.FromInt(1)
-	maxUnavailable := intstr.FromInt(0)
+	maxUnavailable := intstr.FromInt(1)
 
-	deployment := &appsv1.Deployment{
+	deployment := &appsv1.DaemonSet{
 		ObjectMeta: metav1.ObjectMeta{
 			Labels:    labels,
 			Namespace: cr.Namespace,
 			Name:      "submariner",
```

Comment:
Sorry for the late comment, it's something we can handle in a follow up PR as it's unrelated.

Some time ago we mentioned D/S and U/S about changing submariner to submariner-gateway to make it more evident, as some people was confused by what the gateways where. 

I will add a follow up PR to here and the helm charts for uniformity before next release.

Reply:
Submariner-gateway would be apter. @mangelajo ++